src/main/java/com/sonyericsson/jenkins/plugins/bfa/PluginImpl.java

```diff
@@ -629,6 +631,62 @@ public void setSlackFailureCategories(String slackFailureCategories) {
         this.slackFailureCategories = slackFailureCategories;
     }
 
+    /**
+     * Get if causes from the current knowledge base should be included in the JCasC export is enabled.
+     * @return exportCausesToJCasCEnabled - on or off. null == off.
+     */
+    public Boolean getExportCausesToJCasCEnabled() {
+        return exportCausesToJCasCEnabled;
+    }
+
+    /**
+     * Set if this feature is enabled or not. When on, causes will in the JCasC export.
+     *
+     * @param exportCausesToJCasCEnabled on or off. null == off.
+     */
+    @DataBoundSetter
+    public void setExportCausesToJCasCEnabled(Boolean exportCausesToJCasCEnabled) {
+        this.exportCausesToJCasCEnabled = exportCausesToJCasCEnabled;
+    }
+
+    /**
+     * Set the causes.
+     *
+     * @param causes value
+     */
+    @DataBoundSetter
+    public void setCauses(List<FailureCause> causes) {
+        if (knowledgeBase != null) {
+            for (FailureCause cause:causes) {
+                try {
+                    knowledgeBase.saveCause(cause);
+                } catch (Exception e) {
+                    logger.log(Level.WARNING, null, e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Get the causes, used by JCasC to include it in the configuration export.
+     *
+     * @return the causes
+     */
+    public Collection<FailureCause> getCauses() {
+        // The Causes export can be very large so we only include it if enabled.
+        if (exportCausesToJCasCEnabled == null || !exportCausesToJCasCEnabled) {
+            return null;
+        }
+        if (knowledgeBase != null) {
+            try {
+                return knowledgeBase.getCauses();
+            } catch (Exception e) {
+                return null;
```

Comment:
I find it unfortunate there is bare `Exception` declared as a checked exception in the api you are calling, but it is still desirable to log or propagate the exception here.